meshtastic/tests/test_meshtasticd_tcp_interface_ci.py

```
@@ -0,0 +1,251 @@
+    if any(separator in host for separator in ("@", "/", "?", "#", ";")):
+        raise ValueError(
+            f"Invalid {MESHTASTICD_HOST_ENV_VAR}={host!r}: expected HOST[:PORT] only."
+        )
+    if host.count(":") >= 2 and not host.startswith("["):
+        try:
+            ipaddress.IPv6Address(host)
+        except ipaddress.AddressValueError as exc:
+            raise ValueError(
+                f"Invalid {MESHTASTICD_HOST_ENV_VAR}={host!r}: raw IPv6 literals "
+                "with explicit ports must use bracket form like [::1]:4401."
+            ) from exc
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)